src/python/pants/backend/python/util_rules/pex_from_targets_test.py

```diff
@@ -412,11 +412,22 @@ def test_exclude_requirements(
     assert len(pex_request.requirements.req_strings) == (1 if include_requirements else 0)
 
 
-def test_issue_12222(rule_runner: RuleRunner) -> None:
+@pytest.mark.parametrize(
+    "enable_resolves",
+    [
+        False,
+        pytest.param(True, marks=pytest.mark.xfail(reason="#14991 causes this to fail in 2.11.x")),
```

Comment:
This test (which asserts that we don't use the entire lockfile) cannot pass on `main` due to  #14991, but I've validated that it passes on `2.10.x`, and will remove the `xfail` while picking it.